bazel/repository.bzl

```diff
@@ -55,54 +46,38 @@ def opentelemetry_cpp_deps():
 
     # Load abseil dependency(optional)
     maybe(
-        #
-        # Important note:
-        #
-        # The bazel build uses abseil-cpp-20230802.2 here,
-        # while CMake uses more recent versions.
-        #
-        # bazel with abseil-cpp-20240116.2 : build failures in CI
-        # bazel with abseil-cpp-20240116.1 : build failures in CI
-        #
-        # TODO: Fix issue #2619
-        #
         http_archive,
         name = "com_google_absl",
-        sha256 = "7c11539617af1f332f0854a6fb21e296a1b29c27d03f23c7b49d4adefcd102cc",
-        strip_prefix = "abseil-cpp-20230802.2",
+        sha256 = "733726b8c3a6d39a4120d7e45ea8b41a434cdacde401cba500f14236c49b39dc",
+        strip_prefix = "abseil-cpp-20240116.2",
         urls = [
-            "https://github.com/abseil/abseil-cpp/archive/refs/tags/20230802.2.tar.gz",
+            "https://github.com/abseil/abseil-cpp/archive/refs/tags/20240116.2.tar.gz",
         ],
     )
 
-    # Load gRPC dependency
+    # Load this before grpc to pull a newer version
     maybe(
         http_archive,
-        name = "com_github_grpc_grpc_legacy",
-        sha256 = "024118069912358e60722a2b7e507e9c3b51eeaeee06e2dd9d95d9c16f6639ec",
-        strip_prefix = "grpc-1.39.1",
-        urls = [
-            "https://github.com/grpc/grpc/archive/v1.39.1.tar.gz",
-        ],
+        name = "build_bazel_apple_support",
+        sha256 = "c4bb2b7367c484382300aee75be598b92f847896fb31bbd22f3a2346adf66a80",
+        url = "https://github.com/bazelbuild/apple_support/releases/download/1.15.1/apple_support.1.15.1.tar.gz",
```

Comment:
Please clarify with comments why we need this.

Is this for extra helpers, or is it an issue for some platforms like MacOs, or ???

Also, please expand on "Load this before grpc to pull a newer version"

The goal is that people not familiar with bazel (i.e., me) should at least understand the basics, to be able to maintain this code the next time, say, grpc or abseil is upgraded for opentelemetry-cpp.

Likewise for rules_apple.3.5.1

Reply:
done